src/python/pants/backend/jvm/scala_artifact.py

```diff
@@ -14,3 +14,7 @@ class ScalaArtifact(Artifact):
   @property
   def name(self):
     return ScalaPlatform.global_instance().suffix_version(self._base_name)
+
+  @name.setter
+  def name(self, value):
```

Comment:
Is this used anywhere?

Reply:
Ah, sorry I didn't explain this one. The superclass `Artifact` defines `@name.setter`, but originally the subclass did not. So, MyPy complained that the superclass property was `read-write` yet the subclass was `read-only` because this would now violate Liskov substitution.

Really neat that MyPy has this check!

Reply:
Great, thanks!